core/platform/image/gae_image_services.py

```diff
@@ -79,9 +80,36 @@ def compress_image(image_content, scaling_factor):
                     MAX_RESIZE_DIMENSION_PX, float(max(width, height))))
             new_width = int(width * new_scaling_factor)
             new_height = int(height * new_scaling_factor)
-        return images.resize(
-            image_data=image_content,
-            width=min(new_width, MAX_RESIZE_DIMENSION_PX),
-            height=min(new_height, MAX_RESIZE_DIMENSION_PX))
+            new_image_dimensions = (
+                min(new_width, MAX_RESIZE_DIMENSION_PX),
+                min(new_height, MAX_RESIZE_DIMENSION_PX))
+
+            # Thumbnail doesn't work for enlarging images.
+            resized_image = image.resize(new_image_dimensions)
+
+            with io.BytesIO() as output:
+                resized_image.save(output, format=image_format)
+                new_image_content = output.getvalue()
+        elif scaling_factor > 1:
+            new_image_dimensions = (
+                min(new_width, MAX_RESIZE_DIMENSION_PX),
+                min(new_height, MAX_RESIZE_DIMENSION_PX))
+
+            # Thumbnail doesn't work for enlarging images.
+            resized_image = image.resize(new_image_dimensions)
+
+            with io.BytesIO() as output:
+                resized_image.save(output, format=image_format)
+                new_image_content = output.getvalue()
```

Comment:
There's a lot of duplicated code in this block and the one above -- in fact pretty much everything is duplicated. See if you can rewrite it to reduce the duplication.

Reply:
Done